scripts/content-utils/card-tree.ts

```diff
@@ -21,72 +21,135 @@ export type CardLeaf = {
     modifiers?: GameWorldModifier | GameWorldModifier[]
 }
 
+// TODO: Refactor CardTree to use this runtime type guard
+export function isCardTree(node: any): node is CardTree {
+    return node && node.card !== undefined
+}
+
+/**
+ * @param tree The CardTree to recursively turn into an array of cards
+ *
+ * NOTE: The following params should only be used internally by the content utils
+ * to connect CardTrees within other structures, like for example CardSequences.
+ *
+ * For actual content, use `tree.conditions` and `tree.modifiers` instead:
+ * @param _startConditions Internal WorldQuery[] to control when to show this CardTree node. Useful when you want to complete the previous node in a CardSequence before starting the CardTree.
+ * @param _endModifiers Internal GameWorldModifier[] to apply to all CardTree leaf nodes. Useful when you want to complete a CardTree within a CardSequence before moving on to the next node in the CardSequence.
+ * @param _bindRef Internal cardRef to the parent card.
+ */
 export function cardsFromTree(
     tree: Omit<CardTree, "modifiers">,
-    bindRef?: string,
+    _startConditions: WorldQuery[] = [],
+    _endModifiers: GameWorldModifier[] = [],
+    _bindRef?: string,
```

Comment:
I was thinking you might be able to remove the need for passing `_bindRef` (which honestly is a quite obscure name) and instead pass a list of `GameWorldModifiers` that should be applied on both left and right actions. Though if it works, it works. We can refactor later. Content is priority. I am in favour of having a very relaxed approach towards content building functions.

Reply:
This could also be another PR.